docs/deployment.md

```diff
@@ -1,69 +1,93 @@
 
-![Orchestrator deployment](images/orchestrator-deployment.png)
+  The above uses [orchestrator-client](orchestrator-client.md), but you may use the [orchestrator cli](executing-via-command-line.md) if running on a shared-backend setup.
 
-In the above four `orchestrator` services are behind an HTTP load balancer. Only one of them is the _leader_ at any given time; they compete for leadership between themselves and recognize if the leader is non doing its duty.
+##### Promotion rule
 
-To the right: the many topologies polled by `orchestrator`. The leader polls each an every server in those topologies.
+Some servers are better candidate for promotion in the event of failovers. Some servers aren't good picks. Examples:
 
-On top left: the `orchestrator` MySQL backend: a master & three replicas. All 4 services use the same backend database.
+- A server has weaker hardware configuration. You prefer to not promote it.
+- A server is in a remote data center and you don't want to promote it.
+- A server is used as your backup source and has LVM snapshots open at all times. You don't want to promote it.
+- A server has a good setup and is ideal as candidate. You prefer to promote it.
+- A server is OK, and you don't have any particular opinion.
 
-Not shown in this picture (for clarity purposes), but the `orchestrator` backend database and its replicas are themselves one of those topologies
-polled by `orchestrator` It eats its own dogfood.
+You will announce your preference for a given server to `orchestrator` in the following way:
 
-On left, bottom: `orchestrator` CLI is installed on any and all MySQL servers. All these CLI deployments use the very same
-MySQL backend database.
+```
+orchestrator -c register-candidate -i ${::fqdn} --promotion-rule ${promotion_rule}
+```
 
-Not shown in this picture (for clarity purposes), the MySQL servers on these hosts are being polled by `orchestrator` just as those
-on the right.
+Supported promotion rules are:
 
-You may choose to install `orchestrator` on non-MySQL hosts, of course. It has no MySQL dependencies on the deployed host.
+- `prefer`
+- `neutral`
+- `prefer_not`
+- `must_not`
+
+Promotion rules expire after an hour. That's the dynamic nature of `orchestrator`. You will want to setup a cron job that will announce the promotion rule for a server:
+
+```
+*/2 * * * * root "/usr/bin/perl -le 'sleep rand 10' && /usr/bin/orchestrator-client -c register-candidate -i this.hostname.com --promotion-rule prefer"
+```
+
+This setup comes from production environments. The cron entries get updated by `puppet` to reflect the appropriate `promotion_rule`. A server may have `prefer` at this time, and `prefer_not` in 5 minutes from now. Integrate your own service discovery method, your own scripting, to provide with your up-to-date `promotion-rule`.
+
```

Comment:
You can also use the http interface to achieve the same result. That avoids the need direct access to the orchestrator database (for writes) so may be preferred.